rcl_lifecycle/src/com_interface.c

```diff
@@ -325,8 +325,23 @@ rcl_lifecycle_com_interface_fini(
 rcl_ret_t
 rcl_lifecycle_com_interface_publish_notification(
   rcl_lifecycle_com_interface_t * com_interface,
+  const char * transition_label, uint8_t transition_id,
   const rcl_lifecycle_state_t * start, const rcl_lifecycle_state_t * goal)
 {
+  // Get the current system time based on the rcl_clock
+  rcutils_time_point_value_t current_time;
+  rcutils_ret_t time_ret = rcutils_system_time_now(&current_time);
```

Comment:
The clock works, but will the result of `rcutils_system_time_now` serve as the correct time? The transition timestamp is meant to be 

> The time point at which this event occurred.

Which needs to be set during the period of notification. I can't image trying to pass down a `clock` through multiple layers of API so that the user could set the time that way though.

Reply:
this will publish the system time only, but i think `LifecycleNode.clock_interface` should be used. more like, `rck_lifecycle_com_interface` should be initialized with the same clock with lifecycle node. so that if the node is configured with simulation time, we can also publish the simulation time in the message.

Reply:
I definitely agree that this would be the best route to go, but will it be problematic making API changes? As the [`rcl_lifecycle_com_interface`](https://github.com/ros2/rcl/blob/rolling/rcl_lifecycle/include/rcl_lifecycle/data_types.h#L73-#L91) is changed that means all functions initializing `com_interface` must be changed to include a `rcl_clock_t *` argument. This would include adding an argument to [`rcl_lifecycle_state_machine_init`](https://github.com/ros2/rcl/blob/rolling/rcl_lifecycle/src/rcl_lifecycle.c#L194) and all the functions it calls down the line.